online-store/src/app/app.ts

```diff
@@ -0,0 +1,278 @@
+            }
+        }
+    
+        let inputText = document.querySelector('.header-button') as HTMLInputElement;
+        inputText?.addEventListener('click', e => {
+            search();
+        })
+        
+    }
+    search();
+    return newArr;
+}
+
+function viewDisplay(view: string){
+    if (view === 'table') {
+        document.querySelector('.view-mode__table')?.classList.add ('view-active');
+        const productItem: NodeListOf<Element> = document.querySelectorAll('.product-item');
+        productItem.forEach (elem => {
+            if (elem instanceof HTMLDivElement) {
+                elem.style.flexDirection = 'column';
+                elem.style.width = '33.333%';
+            }
+            if (elem.childNodes[1].childNodes[2] instanceof HTMLDivElement) { 
+                elem.childNodes[1].childNodes[2].style.width = '100%';                  
+            }
+        })
+    }
+    if (view === 'list') {
+        document.querySelector('.view-mode__list')?.classList.add ('view-active');
+        const productItem: NodeListOf<Element> = document.querySelectorAll('.product-item');
+        productItem.forEach (elem => {
+            if (elem instanceof HTMLDivElement) {
+                elem.style.flexDirection = 'row';
+                elem.style.width = '100%';
+                if (elem.childNodes[0] instanceof HTMLImageElement) {
+                    elem.childNodes[0].style.maxWidth = '200px';
+                    elem.childNodes[0].style.marginRight = '20px';
+                }
+                if (elem.childNodes[1].childNodes[2] instanceof HTMLDivElement) { 
+                    elem.childNodes[1].childNodes[2].style.width = '50%';                  
+                }
+            }
+        })
+    }
+}
+
+function changeView () {
+    document.querySelector('.view-mode__table')?.addEventListener('click', () => {
+        document.querySelector('.view-mode__list')?.classList.remove ('view-active');
+        viewDisplay('table');
+    });
+
+    document.querySelector('.view-mode__list')?.addEventListener('click', () => {
+        document.querySelector('.view-mode__table')?.classList.remove ('view-active');
+        viewDisplay('list');
+    });
+}
+
+function filtersReset() {
+    const filtersReset = document.querySelector('.filters-reset');
+    const checkboxInput: NodeListOf<HTMLInputElement>  = document.querySelectorAll('.checkbox-input');
+    filtersReset?.addEventListener('click', () => {
+        filtersObj = {
+            price: [minPriceFunc(productsArray), maxPriceFunc(productsArray)],
+            sorting: 'priceAscending'
+        };
+        filtered ();
+        checkboxInput.forEach((el) => {
+            el.checked = false;
+        });
+        slider(productsArray);
+    })
+}
+
+function filtersSave() {
+    const filtersSave = document.querySelector('.filters-save');
+    filtersSave?.addEventListener('click', () => {
+        addFiltersToUrl(filtersObj);        
+    }) 
+}
```

Comment:
Naming Conventions:
The naming convention for functions should follow the camelCase style. So, the function names 'filtersReset' and 'filtersSave' should be 'filterReset' and 'filterSave' respectively.
The type definition for 'checkboxInput' variable is not clear, try using a more descriptive name for it.
```
function filterReset() {
    const resetButton = document.querySelector('.filters-reset');
    const checkboxes: NodeListOf<HTMLInputElement>  = document.querySelectorAll('.checkbox-input');
    if (!resetButton) {
        return;
    }

    resetButton.addEventListener('click', () => {
        filtersObj = {
            price: [minPriceFunc(productsArray), maxPriceFunc(productsArray)],
            sorting: 'priceAscending'
        };
        filtered();
        checkboxes.forEach(checkbox => {
            checkbox.checked = false;
        });
        slider(productsArray);
    });
}

function filterSave() {
    const saveButton = document.querySelector('.filters-save');
    if (!saveButton) {
        return;
    }

    saveButton.addEventListener('click', () => {
        addFiltersToUrl(filtersObj);
    });
}
```